### PR TITLE
Closes #5126 Easing Ms2 migration in projects

### LINKED ISSUE #5126: Improve the update of project dependencies
## Description
The migration of projects to newer version of MapStore2 isn't easy.
With this pull we try to reduce a little bit the difficulties.
Ms2 It's now a folder module in the dependencies of the project. When a user update Ms2 version, just run npm install at project root (better remove node_module folder before) to update all the project dependency.
Different case is for the devDependency, these are not automatically installed and hoisted form MS2 module, so you have the responsibility to sync them with Ms2.
if these haven't been customized in the project you can safely use the update script "npm run updateDevDeps" before to run the npm install process.
The project doesn't need any post install script, Ms2 take care about it.
 

## Acceptance criteria
[ ] Ms2 npm installation works
[ ] Creating a project (custom or standard) works, compile and run.

## Other useful information
In project creation the git repository filed is no mode mandatory

### PR DESCRIPTION
**Pull request by kappu72**
## Description
Improve the update of project dependencies

**Please check if the PR fulfills these requirements**
- [X] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [X] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [X] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**
Project dependencies have to be manually synced with ms2 dependencies on any ms2 version changes

**What is the new behavior?**
To sync dependency just run npm install @prject root.
To sync devDeps "npm run updateDevDeps" before npm install

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

## Other useful information
If you wan't to test the creation of a project before merging, you have to manually move the MapStore2/utility/build/ scripts to <project>/MapStore/utility/build and the MapStore2/package.json
because the Ms2 folder inside the project It's a fresh checkout from git and It hasn't last changes.

### CODE REVIEW

**Comment by kappu72** [comment]
> looks great. 👍
> Can you improve project documentation with the options and the information about this (dev dependencies update script, changes on git URL and other info that changed)

Will be done!

File: docs/developer-guide/project-creation-script.md [hunk 1]
```diff
@@ -69,3 +69,18 @@ In addition to static and templates, the following files from the root MapStore
 * .eslintrc
 * .eslintignore
 * LICENSE.txt
+
+## Update MapStore2 version in a project
+
+To update MapStore2 version enter in the MapStore2 folder and pull desired git version.
```
Comment (mbarto) [inline docs/developer-guide/project-creation-script.md:75]:
```suggestion
To update MapStore2 version enter the MapStore2 folder and pull desired git version.
```

File: docs/developer-guide/project-creation-script.md [hunk 2]
```diff
@@ -69,3 +69,18 @@ In addition to static and templates, the following files from the root MapStore
...
+If MapStore2 devDependencies have been changed you can manually update these in the project package.json file or run the script updateDevDeps
+```sh
+npm run ./updateDevDeps.js
```
Comment (mbarto) [inline docs/developer-guide/project-creation-script.md:78]: maybe node ./updateDevDeps.js ?

Comment (kappu72) [reply docs/developer-guide/project-creation-script.md:78]: No I have added as a script npm updateDevDeps is enough

Comment (mbarto) [reply docs/developer-guide/project-creation-script.md]: ok, but then it should be npm run updateDevDeps I suppose, not npm run ./updateDevDeps.js

Comment (kappu72) [reply docs/developer-guide/project-creation-script.md]: Yes, you are right and I have fixed ;-)

**Pull request merged**